onnxruntime/contrib_ops/rocm/bert/layer_norm.cuh

```diff
@@ -109,6 +109,49 @@ __device__ inline void LayerNorm(
   }
 }
 
+template <typename T, int TPB, int ILP>
+__device__ inline void LayerNormVec(
+    const hipcub::KeyValuePair<T, T>& thread_data, const int ld, const int offset, const T* beta,
+    const T* gamma, const T epsilon, T* output) {
+  // Assuming thread_data is already divided by ld
+  using VecT = aligned_vector<T, ILP>;
+  using BlockReduce = hipcub::BlockReduce<hipcub::KeyValuePair<T, T>, TPB>;
+  __shared__ typename BlockReduce::TempStorage temp_storage;
+  __shared__ T mu;      // mean
+  __shared__ T rsigma;  // 1 / std.dev.
+
+  KeyValuePairSum pair_sum;
+  const auto sum_kv = BlockReduce(temp_storage).Reduce(thread_data, pair_sum);
+
+  if (threadIdx.x == 0) {
+    mu = sum_kv.key;
+    rsigma = Rsqrt(sum_kv.value - mu * mu + epsilon);
+  }
+  __syncthreads();
+
+  if (ILP * threadIdx.x < ld) {
```

Comment:
This line seems not right. I think it should be moved within the for loop (line 137).

Reply:
The for loop(Line 137) is `for (int i = threadIdx.x * ILP; i < ld; i += TPB * ILP)`, the end condition is the condition `if (ILP * threadIdx.x < ld)`.